src/Storefront/Controller/AddressController.php

```diff
@@ -208,47 +263,203 @@ public function addressBook(Request $request, RequestDataBag $dataBag, SalesChan
         return $response;
     }
 
+    #[Route(path: '/account/address/delete/{addressId}', name: 'frontend.account.address.delete', options: ['seo' => false], defaults: ['XmlHttpRequest' => true, '_loginRequired' => true], methods: ['POST'])]
+    public function deleteAddress(string $addressId, Request $request, SalesChannelContext $context, CustomerEntity $customer): Response
+    {
+        if (!Feature::isActive('FEATURE_NEXT_19776')) {
+            $success = true;
+        }
+
+        if (!$addressId) {
+            throw RoutingException::missingRequestParameter('addressId');
+        }
+
+        try {
+            $this->deleteAddressRoute->delete($addressId, $context, $customer);
+
+            if (Feature::isActive('FEATURE_NEXT_19776')) {
+                $this->addFlash(self::SUCCESS, $this->trans('account.addressDeleted'));
+            }
+        } catch (InvalidUuidException|AddressNotFoundException|CannotDeleteDefaultAddressException|CustomerException) {
+            if (Feature::isActive('FEATURE_NEXT_19776')) {
+                $this->addFlash(self::DANGER, $this->trans('account.addressNotDeleted'));
+            } else {
+                $success = false;
+            }
+        }
+
+        if (!Feature::isActive('FEATURE_NEXT_19776')) {
+            return new RedirectResponse($this->generateUrl('frontend.account.address.page', ['addressDeleted' => $success ?? '']));
+        }
+
+        return new RedirectResponse($this->generateUrl('frontend.account.address.page'));
+    }
+
+    #[Route(path: '/widgets/account/address-manager/switch', name: 'frontend.account.addressmanager.switch', options: ['seo' => true], defaults: ['XmlHttpRequest' => true, '_loginRequired' => true, '_loginRequiredAllowGuest' => true], methods: ['POST'])]
+    public function addressManagerSwitch(RequestDataBag $dataBag, SalesChannelContext $context): Response
+    {
+        if (!$dataBag->get(SalesChannelContextService::SHIPPING_ADDRESS_ID)) {
+            $dataBag->remove(SalesChannelContextService::SHIPPING_ADDRESS_ID);
+        }
+
+        if (!$dataBag->get(SalesChannelContextService::BILLING_ADDRESS_ID)) {
+            $dataBag->remove(SalesChannelContextService::BILLING_ADDRESS_ID);
+        }
+
+        $this->contextSwitchRoute->switchContext($dataBag, $context);
+
+        $this->addFlash(self::SUCCESS, $this->trans('account.addressSuccessfulChange'));
+
+        return new RedirectResponse(
+            $this->generateUrl('frontend.checkout.confirm.page')
+        );
+    }
+
+    #[Route(path: '/widgets/account/address-manager', name: 'frontend.account.addressmanager.get', options: ['seo' => true], defaults: ['XmlHttpRequest' => true, '_loginRequired' => true, '_loginRequiredAllowGuest' => true], methods: ['GET'])]
+    public function addressManagerGet(Request $request, SalesChannelContext $context, CustomerEntity $customer): Response
+    {
+        $viewData = new AddressEditorModalStruct();
+        $params = [];
```

Comment:
This seems unnecessary

Update it to provide $viewData->getVars() in ->renderStorefront()